Assets/Scripts/PlayerController.cs

```diff
@@ -0,0 +1,162 @@
+
+        if (!isPaused)
+        {
+            bool isMovingOnGround = (x != 0 || z != 0) && isGrounded;
+            if (isMovingOnGround)
+            {
+                float timeSinceLastStep = Time.time - lastStep;
+                if (timeSinceLastStep > stepFrequency)
+                {
+                    if (stepAudio != null && audioSource != null)
+                    {
+                        lastStep = Time.time;
+                        audioSource.PlayOneShot(stepAudio);
+                    }
+                }
+            }
+            else /* If not on ground or not moving */ {
+                lastStep = 0;
+            }
+
+            int layerMask = 1 << 10; // Layer 10 is Interactable Layers
+
+            bool isThereAHeldObject = isObjHeld && lastHeld != null;
+            if (isThereAHeldObject) // Then use that object
+            {
+                if (Input.GetButtonDown("Interact"))
+                {
+                    lastHeld.Interact();
+                    isObjHeld = lastHeld.IsHeld();
+                }
+                ui.setInteractiveText(lastHeld.GetTooltip());
+            }
+            else // Find nearby object
+            {
+                RaycastHit foundObject;
+                bool isRaycastSuccessful = Physics.Raycast(camera.transform.position, camera.transform.forward, out foundObject, reachDistance, layerMask);
+                if (isRaycastSuccessful)
+                {
+                    GameObject hitObj = foundObject.transform.gameObject;
+                    IInteract io = (IInteract)hitObj.GetComponent("IInteract"); ;
+                    if (Input.GetButtonDown("Interact"))
+                    {
+                        // Interact Before IsHeld Check ALWAYS!!! //
+                        io.Interact();
+                        isObjHeld = io.IsHeld();
+                        if (isObjHeld == true) {
+                            lastHeld = io;
+                        }
+                    }
+                    ui.setInteractiveText(io.GetTooltip());
+                }
+                else
+                {
+                    ui.setInteractiveText("");
+                }
+            }
+            
+            if (Input.GetButtonDown("Jump") && isGrounded)
+            {
+                velocity.y = Mathf.Sqrt(jumpHeight * -2 * gravity);
+            }
+            velocity.y += gravity * Time.deltaTime;
+            controller.Move(velocity * Time.deltaTime);
+
+            controller.Move(move * movementSensitivity * Time.deltaTime);
+            camera.transform.localRotation = Quaternion.Euler(xRotation, 0f, 0f);
+            transform.Rotate(Vector3.up * mouseX);
+        }
+        else {
+            ui.setInteractiveText("");
+        }
+
+        if (Input.GetButtonDown("Escape"))
+        {
+            SetPause(!isPaused);
+        }
+
+    }
+
+
+    public bool IsGamePaused() {
+        return isPaused;
+    }
+    public void SetPause(bool torf) {
```

Comment:
Might want a more descriptive name for "torf"

Reply:
Yeah, torf, short for TrueOrFalse isn't intuitive. Ill fix that right away